### PR TITLE
 Enhance the support for Oracle SQL parsing

### LINKED ISSUE #27067: Support parsing Oracle CREATE OR sql
# Background
Hi community,
This issue is for #26878.

ShardingSphere parser engine helps users parse a SQL to get the AST (Abstract Syntax Tree) and visit this tree to get SQLStatement (Java Object). Currently, we are planning to enhance the support for Oracle SQL parsing in ShardingSphere.

More details:
https://shardingsphere.apache.org/document/current/en/reference/sharding/parse/

# Task
This issue is to support more oracle sql parse, as follows:
```sql
CREATE OR REPLACE TRIGGER lineitems_trigger
  AFTER INSERT OR UPDATE OR DELETE ON lineitems 
  FOR EACH ROW
BEGIN
  IF (INSERTING OR UPDATING)
  THEN
    UPDATE orders SET line_items_count = NVL(line_items_count,0)+1
    WHERE order_id = :new.order_id;
  END IF;
  IF (DELETING OR UPDATING)
  THEN
    UPDATE orders SET line_items_count = NVL(line_items_count,0)-1
    WHERE order_id = :old.order_id;
  END IF;
END;
```

```sql
CREATE OR REPLACE TRIGGER Derived 
BEFORE INSERT OR UPDATE OF Ename ON Emp
```

```sql
CREATE OR REPLACE PACKAGE tests AS
  FUNCTION valid (x NUMBER) RETURN CHAR;
  FUNCTION valid (x DATE) RETURN CHAR;
  PRAGMA RESTRICT_REFERENCES (valid, WNDS);
END;
```

```sql
CREATE OR REPLACE TRIGGER Emp_count
  AFTER DELETE ON employees
DECLARE
  n  INTEGER;
BEGIN
  SELECT COUNT(*) INTO n FROM employees;
  DBMS_OUTPUT.PUT_LINE('There are now ' || n || ' employees.');
END;
```

```sql
CREATE OR REPLACE TRIGGER Employee_permit_changes
  BEFORE INSERT OR DELETE OR UPDATE ON employees
DECLARE
  Dummy             INTEGER;
  Not_on_weekends   EXCEPTION;
  Nonworking_hours  EXCEPTION;
  PRAGMA EXCEPTION_INIT (Not_on_weekends, -4097);
  PRAGMA EXCEPTION_INIT (Nonworking_hours, -4099);
BEGIN
   -- Check for weekends:
 
   IF (TO_CHAR(Sysdate, 'DAY') = 'SAT' OR
     TO_CHAR(Sysdate, 'DAY') = 'SUN') THEN
       RAISE Not_on_weekends;
   END IF;
 
  -- Check for work hours (8am to 6pm):
 
  IF (TO_CHAR(Sysdate, 'HH24') < 8 OR
    TO_CHAR(Sysdate, 'HH24') > 18) THEN
      RAISE Nonworking_hours;
  END IF;
 
EXCEPTION
  WHEN Not_on_weekends THEN
    Raise_application_error(-20324,'Might not change '
      ||'employee table during the weekend');
  WHEN Nonworking_hours THEN
    Raise_application_error(-20326,'Might not change '
     ||'emp table during Nonworking hours');
END;
```

# Process
1. First confirm that this is a correct oracle sql syntax, if not please ignore;
2. Compare SQL definitions in Oficial SQL Doc and ShardingSphere SQL Doc;
3. If there is any difference in ShardingSphere SQL Doc, please correct them by referring to the Official SQL Doc;
4. Run mvn install the current_file_module;
5. Check whether there are any exceptions. If indeed, please fix them. (Especially xxxVisitor.class);
6. Add new corresponding SQL case in SQL Cases and expected parsed result in Expected Statment XML;
7. Run SQLParserParameterizedTest to make sure no exceptions.

# Relevant Skills
1. Master JAVA language
2. Have a basic understanding of Antlr `g4` file
3. Be familiar with Oracle SQLs

### PR DESCRIPTION
**Pull request by zhuruixuan**
Fixes #27067.

Changes proposed in this pull request:
  -Enhance the support for Oracle SQL parsing in ShardingSphere

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: test/it/parser/src/main/resources/sql/supported/ddl/create-trigger.xml [hunk 1]
```diff
@@ -50,4 +50,47 @@
...
+    <sql-case id="create_trigger_with_exceptionInit_pragma" value="CREATE OR REPLACE TRIGGER Employee_permit_changes
+  BEFORE INSERT OR DELETE OR UPDATE ON employees
+DECLARE
+  Dummy             INTEGER;
+  Not_on_weekends   EXCEPTION;
+  Nonworking_hours  EXCEPTION;
+  PRAGMA EXCEPTION_INIT (Not_on_weekends, -4097);
+  PRAGMA EXCEPTION_INIT (Nonworking_hours, -4099);
+BEGIN
+
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/sql/supported/ddl/create-trigger.xml:77]: Please remove this useless blank line.

File: test/it/parser/src/main/resources/sql/supported/ddl/create-trigger.xml [hunk 2]
```diff
@@ -50,4 +50,47 @@
         BEGIN
             :NEW.ID := Contacts_Seq.NEXTVAL;
         END;" db-types="Oracle" />
+    <sql-case id="create_trigger_with_dataManipulationLanguage_statement" value="CREATE OR REPLACE TRIGGER lineitems_trigger
+  AFTER INSERT OR UPDATE OR DELETE ON lineitems
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/sql/supported/ddl/create-trigger.xml:54]: Is it possible to increase the indentation?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: test/it/parser/src/main/resources/sql/supported/ddl/create-trigger.xml (hunk 2 above)
Comment (zhuruixuan) [reply test/it/parser/src/main/resources/sql/supported/ddl/create-trigger.xml]: I have added appropriate indentation to improve the readability of the code

> Is it possible to increase the indentation?

**Review by strongduanmu — APPROVED** [review]
LGTM, merged.

**Pull request merged**